dlc/src/main/scala/org/bitcoins/dlc/verify/DLCSignatureVerifier.scala

```diff
@@ -135,31 +143,41 @@ object DLCSignatureVerifier extends BitcoinSLogger {
       localIsInitiator: Boolean,
       offerFundingInputs: Vector[DLCFundingInput],
       acceptFundingInputs: Vector[DLCFundingInput]): Boolean = {
-    val (remoteTweak, remoteFundingInputs) = if (localIsInitiator) {
-      (offerFundingInputs.length, acceptFundingInputs)
-    } else {
-      (0, offerFundingInputs)
-    }
+    val fundingInputs = offerFundingInputs ++ acceptFundingInputs
+
+    val serialIdMap =
+      fundingInputs
+        .map(input => input.outPoint -> input.inputSerialId)
+        .toMap
+
+    val serialIds = fundingInputs.map(_.inputSerialId).sorted
+
+    val remoteFundingInputs =
+      if (localIsInitiator) acceptFundingInputs
+      else offerFundingInputs
 
     val psbt = PSBT.fromUnsignedTxWithP2SHScript(fundingTx)
 
     fundingSigs.zipWithIndex
       .foldLeft(true) { case (ret, ((outPoint, witness), index)) =>
-        val idx = index + remoteTweak
+        val serialId = serialIdMap(outPoint)
+        val idx = serialIds.indexOf(serialId)
         if (ret) {
           if (psbt.transaction.inputs(idx).previousOutput != outPoint) {
             logger.error("Adding signature for incorrect input")
 
             false
           } else {
-            val fundingInput = remoteFundingInputs(index)
-
-            psbt
-              .addUTXOToInput(fundingInput.prevTx, idx)
-              .addFinalizedScriptWitnessToInput(fundingInput.scriptSignature,
-                                                witness,
-                                                idx)
-              .finalizeInput(idx) match {
+            Try {
```

Comment:
not sure I follow why this try is needed?

Reply:
With invalid signatures this would sometimes throw NoSuchElement or IndexOutOfBounds exceptions